k4MarlinWrapper/src/components/EDM4hep2Lcio.cpp

```diff
@@ -212,6 +212,30 @@ void EDM4hep2LcioTool::convertReconstructedParticles(
   lcio_event->addCollection(conv_recops, lcio_coll_name);
 }
 
+// Transfer info from EDM4hep EventHeader to LCIO event
+void EDM4hep2LcioTool::convertEventHeader(const std::string& e4h_coll_name, lcio::LCEventImpl* lcio_event) {
+  // ReconstructedParticles handle
+  DataHandle<edm4hep::EventHeaderCollection> header_handle{e4h_coll_name, Gaudi::DataHandle::Reader, this};
+  const auto                                 header_coll = header_handle.get();
+
+  const auto event_n      = header_coll->eventNumber();
+  const auto run_n        = header_coll->runNumber();
+  const auto timestamp    = header_coll->timeStamp();
+  const auto event_weight = header_coll->weight();
+
+  // the collection returns vectors but they should be of length 1
+  if (event_n.size() != 1 || run_n.size() != 1) {
+    // TODO: fail harder?
+    error() << "Malformed EventHeader, multiple entries for event number, run number, timestamp or weight!" << endmsg;
+    return;
+  }
+
+  lcio_event->setEventNumber(event_n[0]);
+  lcio_event->setRunNumber(run_n[0]);
+  lcio_event->setTimeStamp(timestamp[0]);
```

Comment:
At the moment there are only checks for eventnumber and runnumber, so timestamp and event_weight might be empty.

Reply:
Thanks for spotting this